### PR TITLE
Filtermenu buttons template

### PR DESCRIPTION
**Pull request by svdimitr**

### CODE REVIEW

File: components/grid/templates/filter.md [hunk 1]
```diff
@@ -305,6 +308,97 @@ For an example with the CheckboxList Filter, see the [Custom Data]({%slug grid-c
 
 ![Custom Filter Menu Template with Checkboxes](images/custom-filter-menu-checkboxes.png)
 
+## Filter Menu Buttons Template
+
+By default, the Filter Menu renders the `Filter` and `Clear` buttons. You can customize or remove them entirely by using the `FilterMenuButtonsTemplate` tag. 
+
+The template receives a `context` of type `FilterMenuTemplateContext` that provides the following members:
+
+* `FilterDescriptor` - the object that describes the column filter. By default it has two filters with the type and name of the field, and you can add more to its `FilterDescriptors` collection, or change its `LogicalOperator` from the default `AND`. You can use the filter descriptor to create a custom button that applies a predefined filter.
```
Comment (yordan-mitev) [inline components/grid/templates/filter.md:317]: Replace the pronouns with the actual nouns to improve the readability:
- what does "it" refer to ("By default it has two filters...")?
- "its `FilterDescriptors`",  "its `LogicalOperator`"

```suggestion
* `FilterDescriptor`—the object that describes the column filter. By default, the column filter has two filters: one for the type and another for the name of the field. You can modify the column filter by:
   * Adding more filters to the `FilterDescriptors` collection.
   * Changing the `LogicalOperator` (`AND` by default).
   * Using the `FilterDescriptor` to create a custom button that applies a predefined filter. 
```

File: components/grid/templates/filter.md [hunk 2]
```diff
@@ -305,6 +308,97 @@ For an example with the CheckboxList Filter, see the [Custom Data]({%slug grid-c
...
 ![Custom Filter Menu Template with Checkboxes](images/custom-filter-menu-checkboxes.png)
 
+## Filter Menu Buttons Template
+
+By default, the Filter Menu renders the `Filter` and `Clear` buttons. You can customize or remove them entirely by using the `FilterMenuButtonsTemplate` tag. 
+
+The template receives a `context` of type `FilterMenuTemplateContext` that provides the following members:
+
+* `FilterDescriptor` - the object that describes the column filter. By default it has two filters with the type and name of the field, and you can add more to its `FilterDescriptors` collection, or change its `LogicalOperator` from the default `AND`. You can use the filter descriptor to create a custom button that applies a predefined filter. 
+* `FilterAsync` - applies the defined filters in the Filter Menu to the Grid component.
```
Comment (yordan-mitev) [inline components/grid/templates/filter.md:318]:
```suggestion
* `FilterAsync`—applies the filters defined in the Filter Menu to the Grid component.
```

File: components/grid/templates/filter.md [hunk 3]
```diff
@@ -305,6 +308,97 @@ For an example with the CheckboxList Filter, see the [Custom Data]({%slug grid-c
...
+* `ClearFilterAsync` - clears the applied filters.
```
Comment (yordan-mitev) [inline components/grid/templates/filter.md:319]:
```suggestion
* `ClearFilterAsync`—clears the applied filters.
```

File: components/grid/templates/filter.md [hunk 4]
```diff
@@ -305,6 +308,97 @@ For an example with the CheckboxList Filter, see the [Custom Data]({%slug grid-c
...
+
+>caption Implement a button with predefined filter.
```
Comment (yordan-mitev) [inline components/grid/templates/filter.md:321]:
```suggestion
>caption Implementing a button with a predefined filter.
```

File: components/grid/templates/filter.md [hunk 5]
```diff
@@ -305,6 +308,100 @@ For an example with the CheckboxList Filter, see the [Custom Data]({%slug grid-c
 
 ![Custom Filter Menu Template with Checkboxes](images/custom-filter-menu-checkboxes.png)
 
+## Filter Menu Buttons Template
+
+By default, the Filter Menu renders the `Filter` and `Clear` buttons. You can customize or remove them entirely by using the `FilterMenuButtonsTemplate` tag.
```
Comment (dimodi) [inline components/grid/templates/filter.md:313]:
```suggestion
By default, the Filter Menu renders `Filter` and `Clear` buttons. You can customize or remove them entirely by using the `FilterMenuButtonsTemplate` tag. 
```

File: components/grid/templates/filter.md [hunk 6]
```diff
@@ -305,6 +308,100 @@ For an example with the CheckboxList Filter, see the [Custom Data]({%slug grid-c
...
+The template receives a `context` of type `FilterMenuTemplateContext` that provides the following members:
+
+* `FilterDescriptor`—the object that describes the column filter. By default, the column filter has two filters: one for the type and another for the name of the field. You can modify the column filter by:
+   * Adding more filters to the `FilterDescriptors` collection.
+   * Changing the `LogicalOperator` (`AND` by default).
+   * Using the `FilterDescriptor` to create a custom button that applies a predefined filter. 
+* `FilterAsync`—applies the filters defined in the Filter Menu to the Grid component.
+* `ClearFilterAsync`—clears the applied filters.
+
+>caption Implementing a button with a predefined filter.
```
Comment (dimodi) [inline components/grid/templates/filter.md:324]: Not sure what "predefined filter" is.

```suggestion
>caption Using custom filter menu buttons
```

**Review by dimodi — APPROVED** [review]
* Use a table for the `context` properties.
* Minor suggestions.

**Pull request merged**